homeassistant/components/sabnzbd/__init__.py

```diff
@@ -1,319 +1,119 @@
+        sab_api_data = SabnzbdApiData(sab_api)
 
-    async_track_time_interval(hass, async_update_sabnzbd, UPDATE_INTERVAL)
+        async def async_service_handler(service: ServiceCall) -> None:
+            """Handle service calls."""
+            if service.service == SERVICE_PAUSE:
+                await sab_api_data.async_pause_queue()
+            elif service.service == SERVICE_RESUME:
+                await sab_api_data.async_resume_queue()
+            elif service.service == SERVICE_SET_SPEED:
+                speed = service.data.get(ATTR_SPEED)
+                await sab_api_data.async_set_queue_speed(speed)
 
+        hass.services.async_register(
+            DOMAIN, SERVICE_PAUSE, async_service_handler, schema=vol.Schema({})
+        )
 
-@callback
-def async_request_configuration(hass, config, host, web_root):
-    """Request configuration steps from the user."""
+        hass.services.async_register(
+            DOMAIN, SERVICE_RESUME, async_service_handler, schema=vol.Schema({})
+        )
 
-    # We got an error if this method is called while we are configuring
-    if host in _CONFIGURING:
-        configurator.async_notify_errors(
-            hass, _CONFIGURING[host], "Failed to register, please try again."
+        hass.services.async_register(
+            DOMAIN, SERVICE_SET_SPEED, async_service_handler, schema=SPEED_LIMIT_SCHEMA
         )
 
-        return
+        async def async_update_sabnzbd(now):
+            """Refresh SABnzbd queue data."""
+            try:
+                await sab_api.refresh_data()
+                async_dispatcher_send(hass, SIGNAL_SABNZBD_UPDATED, None)
+            except SabnzbdApiException as err:
+                _LOGGER.error(err)
 
-    async def async_configuration_callback(data):
-        """Handle configuration changes."""
-        api_key = data.get(CONF_API_KEY)
-        sab_api = SabnzbdApi(
-            host, api_key, web_root=web_root, session=async_get_clientsession(hass)
-        )
-        if not await async_check_sabnzbd(sab_api):
-            return
-
-        def success():
-            """Signal successful setup."""
-            conf = load_json(hass.config.path(CONFIG_FILE))
-            conf[host] = {CONF_API_KEY: api_key}
-            save_json(hass.config.path(CONFIG_FILE), conf)
-            req_config = _CONFIGURING.pop(host)
-            configurator.request_done(hass, req_config)
-
-        hass.async_add_job(success)
-        async_setup_sabnzbd(hass, sab_api, config, config.get(CONF_NAME, DEFAULT_NAME))
-
-    _CONFIGURING[host] = configurator.async_request_config(
-        hass,
-        DEFAULT_NAME,
-        async_configuration_callback,
-        description="Enter the API Key",
-        submit_caption="Confirm",
-        fields=[{"id": CONF_API_KEY, "name": "API Key", "type": ""}],
-    )
+        async_track_time_interval(hass, async_update_sabnzbd, UPDATE_INTERVAL)
+
+    async_setup_sabnzbd(hass, sab_api)
```

Comment:
There is no need this needs to be a separate method.

Reply:
removed method